lib/Onyx.js

```diff
@@ -1665,12 +1669,20 @@ function init({
     // Initialize all of our keys with data provided then give green light to any pending connections
     Promise.all([addAllSafeEvictionKeysToRecentlyAccessedList(), initializeWithDefaultKeyStates()]).then(deferredInitTask.resolve);
 
-    if (shouldSyncMultipleInstances && _.isFunction(Storage.keepInstancesSync)) {
+    if (shouldSyncMultipleInstances) {
         Storage.keepInstancesSync((key, value) => {
             cache.set(key, value);
             keyChanged(key, value);
         });
     }
+
+    return isUsingFallback;
+}
+
+function isUsingFallbackStorageProvider() {
```

Comment:
I'm a little confused about why we have `isUsingFallbackStorageProvider()` and also `Storage.init()` that _seems like_ it would answer the same question.

Reply:
It's answering the same question, yes. For `Onyx.init` we would use the return value at app startup and react with showing a banner or disabling certain features, or whatever.

Th `isUsingFallbackStorageProvider` was just an extra function to check for this after initialization phase.

Also, since i designed this new storage interface in a way that you can potentially also use another fallback storage and/or add/remove providers manually, we can use this to check if the "main" storage provider is working after the replacement.

Does that make sense? 😅

Reply:
Yes! It makes sense. I am envisioning an alternate approach where if the storage fails then we just set a boolean like `Onyx.isMemoryOnly()` and the app developers can check to see if storage is working or not.